frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -235,15 +240,18 @@ pub mod pallet {
             let force_new_era = Self::force_era().eq(&Forcing::ForceNew);
             let blocks_per_era = T::BlockPerEra::get();
             let previous_era = Self::current_era();
+            let next_era_starting_block = Self::next_era_starting_block();
 
             // Value is compared to 1 since genesis block is ignored
-            if now % blocks_per_era == BlockNumberFor::<T>::from(1u32)
+            if now == next_era_starting_block
```

Comment:
This is good!

About the question for runtime upgrade - try to do the following:
* run a local node without your code
* build a new runtime which includes your code
* do runtime upgrade
* check what happens with era changes

Reply:
This seems quite obvious now that you've written the steps out :)